SharedSource/Development/UnitTesting/NUnit/NUnitExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System;
+using NUnit.Framework.Constraints;
+using Remotion.Utilities;
+
+namespace Remotion.Development.UnitTesting.NUnit
+{
+  public static partial class NUnitExtensions
+  {
+    public static EqualConstraint ArgumentOutOfRangeExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName, int actualValue)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentOutOfRangeException (message: message, paramName: paramName, actualValue: actualValue).Message);
+    }
+
+    public static ContainsConstraint ArgumentExceptionMessageContains (this ConstraintExpression constraintExpression, string message, string paramName)
+    {
+      return constraintExpression.With.Message.Contains (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static EqualConstraint ArgumentExceptionMessageEqualTo (this ConstraintExpression constraintExpression, string message, string paramName)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.EqualTo (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static RegexConstraint ArgumentExceptionMessageMatches (this ConstraintExpression constraintExpression, string message, string paramName)
+    {
+      AssertThatMessageContainsWhitespaces (message: message);
+      AssertThatParameterDoesNotContainWhitespaces (paramName: paramName);
+      return constraintExpression.With.Message.Matches (new ArgumentException (message: message, paramName: paramName).Message);
+    }
+
+    public static EndsWithConstraint ArgumentNullExceptionMessageEndsWith (this ConstraintExpression constraintExpression, string paramName)
```

Comment:
Probably should change this to ArgumentNullExceptionMessageEqualTo with corresonding semantics. Don't need to pass a message. still should consider, if this is really needed. we have another assertion somewhere where we only go for the parameter name I think? maybe unify towards that variant instead?

Reply:
`ArgumentNullExceptionMessageEndsWith` checks if the parameter name is correct, therefore it can also be replaced with the new method which creates a new message containing only the parameter name string.